sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -618,6 +618,22 @@ object DateTimeUtils {
     instantToMicros(resultTimestamp.toInstant)
   }
 
+  /**
+   * Add the date and the interval's months and days.
+   * Returns a date value, expressed in days since 1.1.1970.
+   *
+   * @throws DateTimeException if the result exceeds the supported date range
+   * @throws IllegalArgumentException if the interval has `microseconds` part
+   */
+  def dateAddInterval(
+     start: SQLDate,
+     interval: CalendarInterval): SQLDate = {
+    require(interval.microseconds == 0,
+      "Cannot add hours, minutes or seconds, milliseconds, microseconds to a date")
+    val ld = LocalDate.ofEpochDay(start).plusMonths(interval.months).plusDays(interval.days)
```

Comment:
Actually, the result depends on the order of `plusMonths()` and `plusDays()`. @yaooqinn Did you make the choice intentionally? I am asking you because adding days and months can be much cheaper.

Reply:
Yes，here we are follow the previous behavior of using timestamp + interval

Reply:
I see, thanks. It would be nice to document such behavior of this function and timestampAddInterval somewhere. It is not obvious that we add month then days and then micros. The order could be opposite.

Reply:
FYI, in snowflake `internal '1 month 1 day'` is different from `internal '1 day 1 month'`. We should at least document our own behavior.

Reply:
I will make time for that PR.

FYI, without year-month/day-time specific interval type implementation, the operation below looks odd.
```sql
spark-sql> select date '2020-04-30' + interval '1 day 1 month' - interval '1 day 1 month';
2020-04-29
```